README.md

```diff
@@ -106,3 +106,212 @@ jobs:
           name: yamllint-logfile
           path: ${{ steps.yaml-lint.outputs.logfile }}
 ```
+
+## Using self-hosted runner with containers
```

Comment:
Do you mind creating a new file self-hosted-runner.md with the content for self-hosted and create a markdown link from README.md to it? I'd rather keep the readme as short as possible.

Reply:
Hi, I have updated the PR and created the **self-hosted-runner.md** file as you requested.